src/couch_stats/src/couch_stats_resource_tracker.erl

```
@@ -0,0 +1,1014 @@
+        changes_returned = ChangesReturned,
+        ioq_calls = IoqCalls
+    } = Rctx,
+    PidRef = {term_to_flat_json(Pid), term_to_flat_json(Ref)},
+    MFA = case MFA0 of
+        {M0, F0, A0} ->
+            M = atom_to_binary(M0),
+            F = atom_to_binary(F0),
+            A = integer_to_binary(A0),
+            <<M/binary, ":", F/binary, "/", A/binary>>;
+        MFA0 when is_list(MFA0) ->
+            MFA0;
+        undefined ->
+            null;
+        OtherMFA ->
+            throw({error, {unexpected, OtherMFA}})
+    end,
+    From = case From0 of
+        {Parent, ParentRef} ->
+            {term_to_flat_json(Parent), term_to_flat_json(ParentRef)};
+        undefined ->
+            null
+    end,
+    State = case State0 of
+        alive ->
+            alive;
+        {down, Reason0}  ->
+            Reason = term_to_flat_json(Reason0),
+            <<"down: ", Reason/binary>>
+    end,
+    Nonce = case Nonce0 of
+        undefined ->
+            null;
+        Nonce0 ->
+            list_to_binary(Nonce0)
+    end,
+    #{
+        updated_at => TP,
+        started_at => TInit,
+        pid_ref => PidRef,
+        mfa => MFA,
+        nonce => Nonce,
+        from => From,
+        dbname => DbName,
+        username => UserName,
+        db_open => DbOpens,
+        docs_read => DocsRead,
+        js_filter => JSFilters,
+        js_filter_error => JSFilterErrors,
+        js_filtered_docs => JSFilteredDocss,
+        rows_read => RowsRead,
+        state => State,
+        type => term_to_flat_json({type, Type}),
+        kp_nodes => KpNodes,
+        kv_nodes => KvNodes,
+        btree_folds => ChangesProcessed,
+        changes_returned => ChangesReturned,
+        ioq_calls => IoqCalls
+    }.
+
+
+convert_mfa({M0, F0, A0}) ->
+    M = atom_to_binary(M0),
+    F = atom_to_binary(F0),
+    A = integer_to_binary(A0),
+    <<M/binary, ":", F/binary, "/", A/binary>>;
+convert_mfa(undefined) ->
```

should we handle a case when it is something else.

```
convert_mfa(Else) -> 
    list_to_binary(lists:flatten(io_lib:format("{unexpected, ~p}", [Else]))).

---

maybe not. Let it crash.